File: HelpDeskBackend/Models/Ticket.cs
using System;
using System.Collections.Generic;

namespace HelpDeskBackend.Models;

public partial class Ticket
{

    public int Id { get; set; }

    public string? Description { get; set; }

    public string? Resolution { get; set; }

    public bool? IsClosed { get; set; }

    [System.Text.Json.Serialization.JsonIgnore]
    public virtual ICollection<Bookmark> Bookmarks { get; set; } = new List<Bookmark>();
}

File: HelpDeskFrontend/src/app/ticket.ts
// To parse this data:
//
//   import { Convert, Ticket } from "./file";
//
//   const ticket = Convert.toTicket(json);

import { Bookmark } from "./bookmark";

export interface Ticket {
    id:          number;
    description: string;
    resolution:  string;
    isClosed:    boolean;
    bookmarks:   any[];
}

// Converts JSON strings to/from your types
export class Convert {
    public static toTicket(json: string): Ticket {
        return JSON.parse(json);
    }

    public static ticketToJson(value: Ticket): string {
        return JSON.stringify(value);
    }
}
